vsts/pipelines/templates/_releaseJobTemplate.yml

```diff
@@ -57,8 +57,8 @@ jobs:
     skipComponentGovernanceDetection: true
 
   steps:
-  - task: DownloadBuildArtifacts@0
-    displayName: 'Download Artifacts for release'
+  - task: DownloadPipelineArtifact@2
```

Comment:
nit: I would prefer to keep the task version same in a specific job/task template

Reply:
These are different task types actually